src/notebooksyncagent.cpp

```diff
@@ -807,6 +794,19 @@ void NotebookSyncAgent::requestFinished(Request *request)
     request->deleteLater();
 
     if (mRequests.isEmpty()) {
+        if (!mSentUids.isEmpty()) {
+            const QList<Reader::CalendarResource> &resources = mReceivedCalendarResources;
+            for (const Reader::CalendarResource &resource : resources) {
+                if (mSentUids.contains(resource.href) && resource.etag.isEmpty()) {
+                    // Asked for a resource etag but didn't get it.
+                    mFailingUploads.insert(resource.href, QByteArray("Cannot retrieve etag."));
```

Comment:
nitpick: "Unable to retrieve etag" maybe, as "cannot" suggests that it won't be able to in future either, which may not be true.

Reply:
Sorry, I've forgotten about it. It's corrected.